src/WebJobs.Script/Rpc/Configuration/WorkerConfigFactory.cs

```diff
@@ -220,5 +223,37 @@ internal string GetExecutablePathForJava(string defaultExecutablePath)
                 return Path.GetFullPath(Path.Combine(javaHome, "bin", defaultExecutablePath));
             }
         }
+
+        internal string HydrateWorkerPath(string workerPath, string language)
+        {
+            if (string.IsNullOrEmpty(workerPath))
+            {
+                return null;
+            }
+
+            string os = _systemRuntimeInformation.GetOSPlatform().ToString();
+
+            string architecture = _systemRuntimeInformation.GetOSArchitecture().ToString();
+            string version = Environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeVersionSettingName);
+            if (string.IsNullOrEmpty(version))
+            {
+                version = GetWorkerRuntimeDefaultVersion(language);
+            }
+
+            return workerPath.Replace("@os", os)
+                             .Replace("@architecture", architecture)
+                             .Replace($"{{{LanguageWorkerConstants.FunctionWorkerRuntimeVersionSettingName}}}", version);
+        }
+
+        internal string GetWorkerRuntimeDefaultVersion(string language)
+        {
+            if (string.Equals(language, LanguageWorkerConstants.PythonLanguageWorkerName,
+                StringComparison.OrdinalIgnoreCase))
+            {
```

Comment:
We should not be adding defaults specific to a language in the runtime. This should be coming from worker.config.json as well.